blocks/api/test/registration.js

```diff
@@ -180,6 +180,135 @@ describe( 'blocks', () => {
 			} );
 		} );
 
+		it( 'should validate the icon', () => {
+			const blockType = {
+				settingName: 'settingValue',
+				save: noop,
+				category: 'common',
+				title: 'block title',
+				icon: { chicken: 'ribs' },
+			};
+			const block = registerBlockType( 'core/test-block-icon-normalize-element', blockType );
+			const errorMsg =
+				'The icon passed is invalid.' +
+				'The icon should be a string with a Slug of the DashIcon or an element (or function returning an element) if you choose to render your own SVG.' +
+				'An object can also be passed, in this case, icon, as specified above, should be included in the src property.';
+			expect( console ).toHaveErroredWith( errorMsg );
```

Comment:
Minor: Not sure what we'd want to promote here, but with such complex message I'd lean more toward `expect( console ).toHaveErrored()` or, to avoid _real_ errors, maybe `toHaveErroredWith( expect.stringMatching( /* ... */ ) )`

Reply:
Thinking about it expect( console ).toHaveErrored() should be enough for testing purposes and allows to change the message without the need to update the tests.